synapse/storage/data_stores/main/__init__.py

```diff
@@ -473,6 +473,24 @@ def _generate_user_daily_visits(txn):
             "generate_user_daily_visits", _generate_user_daily_visits
         )
 
+    def get_user(self, user):
```

Comment:
I think `get_user_by_id` does what you need here (or could do with a minor update)

Reply:
Actually I'd like to integrate also profile information like display name and picture URL as this makes the client much simpler and admins may see this data anyway. But right now I'm not sure how to achieve this in the best way.

Reply:
easiest way is probably just to make a separate call to `store.get_profileinfo`

Reply:
Can it also be done using "LEFT JOIN" in the SQL statement? I guess that should yield the best performance.

Reply:
yeesss, but this feels like premature optimisation: it adds complexity to the code (more storage methods) for a tiny performance gain